src/telegram_bot/bot.py

```diff
@@ -63,21 +62,64 @@ async def save_to_s3(message):
     os.remove(image_path)
     return s3_path
 
-@dp.message_handler(state=QStates.find, content_types=['photo'])
-async def process_find(message: types.Message, state: FSMContext):
-    additional_info = message.to_python().get("caption", "")
-    s3_path = await save_to_s3(message)
-    kafka_message = to_message(message.from_user.id, s3_path, additional_info)
-    kafka_producer.send(value=kafka_message, key=message.from_user.id, topic='find_cat')
-    await message.answer("Thanks! We notify you when we'll get any news")
+def point_to_quadkey(lon: float, lat: float, zoom: int = 16) -> str:
+    tile = mercantile.tile(lon, lat, zoom)
+    return mercantile.quadkey(tile)
+
+@dp.message_handler(is_media_group=True, content_types=types.ContentType.ANY, state=[RStates.find, RStates.saw])
+async def save_album_to_s3(message: types.Message, album: list, state: FSMContext, cat_name: str):
+    """This handler will receive a complete album of any type."""
+    # media_group = types.MediaGroup()
 
-@dp.message_handler(state=QStates.saw, content_types=['photo'])
-async def process_saw(message: types.Message, state: FSMContext):
-    additional_info = message.to_python().get("caption", "")
+    s3_paths = []
+    for message in album:
+        if message.photo:
+            s3_path = await save_to_s3(message)
+            s3_paths.append(s3_path)
+
+    await state.set_state(RStates.ask_extra_info)
+    await state.update_data(s3_paths=s3_paths, cat_name=cat_name)
+    await message.answer("Please write some extra info about this cat")
+
+@dp.message_handler(content_types=['photo'], state=[RStates.find, RStates.saw])
+async def save_photo_to_s3(message: types.Message, state: FSMContext, cat_name: str):
     s3_path = await save_to_s3(message)
-    kafka_message = to_message(message.from_user.id, s3_path, additional_info)
-    kafka_producer.send(value=kafka_message, key=message.from_user.id, topic='saw_cat')
-    await message.answer("Thank you !!!")
+    await state.set_state(RStates.ask_extra_info)
+    await state.update_data(s3_paths=[s3_path], cat_name=cat_name)
+    await message.answer("Please write some extra info about this cat")
+
+
+@dp.message_handler(state=RStates.ask_extra_info, content_types=['text'])
+async def get_extra_info_and_send(message: types.Message, state: FSMContext):
+    await state.update_data(additional_info=message.text)
+
+    cat_data = await state.get_data()
+    cat_data['quadkey'] = 'no quadkey'
```

Comment:
if you don't have value lets it be None in code and in db

Reply:
ok

Reply:
cat_data['quadkey'] = None